### PR TITLE
CH-2677: As a Lift JS API consumer I want to easily get segments base…

### PR DESCRIPTION
**Pull request by marciocm**
…d on a tracking ID.
- Added trigger to segmentCache.store();
- Added qunit test.

### CODE REVIEW

File: acquia_lift_profiles/js/acquia_lift_profiles.js [hunk 1]
```diff
@@ -51,6 +51,7 @@ var _tcwq = _tcwq || [];
             }
           }
         }
+        $(document).trigger('retrievedSegments', [segments]);
```
Comment (katbailey) [inline acquia_lift_profiles/js/acquia_lift_profiles.js]: Can we rename this event to acquiaLiftStoredSegments? That namespaces it and also matches what is actually happening here, which is that we are storing segments in the cache.

File: acquia_lift_profiles/qunit/tests.js [hunk 2]
```diff
@@ -177,6 +177,28 @@ QUnit.asyncTest("Get context values no cache", function( assert ) {
 
 });
 
+QUnit.asyncTest("Trigger retrievedSegments", function( assert ) {
+  expect(5);
+  var contextResult = Drupal.personalize.visitor_context.acquia_lift_profiles_context.getContext({'segment1':'segment1', 'segment2':'segment2'});
+  assert.ok(contextResult instanceof Promise);
+  var cached = Drupal.personalize.visitor_context_read('segment1', 'acquia_lift_profiles_context');
```
Comment (katbailey) [inline acquia_lift_profiles/qunit/tests.js]: This is confusing me. Maybe you could add some comments to explain what is being tested here? Should there perhaps be an assertion that cached does not contain any segments at this point?

File: acquia_lift_profiles/qunit/tests.js [hunk 3]
```diff
@@ -177,6 +177,28 @@ QUnit.asyncTest("Get context values no cache", function( assert ) {
...
+  QUnit.stop();
+  Promise.all([contextResult]).then(function (loadedContexts) {
+    QUnit.start();
+    cached = Drupal.personalize.visitor_context_read('segment1', 'acquia_lift_profiles_context');
```
Comment (katbailey) [inline acquia_lift_profiles/qunit/tests.js]: Do we even need to call this here? When the promise resolves it stores the segments in the cache and triggers the event, right? So I think we just need the call to QUnit.start() in here?

File: acquia_lift_profiles/js/acquia_lift_profiles.js (hunk 1 above)
Comment (marciocm) [inline acquia_lift_profiles/js/acquia_lift_profiles.js]: Renamed.

File: acquia_lift_profiles/qunit/tests.js (hunk 2 above)
Comment (marciocm) [inline acquia_lift_profiles/qunit/tests.js]: Modified the test to only make sure acquiaLiftStoredSegments is being triggered.

**Pull request closed without merging**